application/models/post.go

```diff
@@ -511,6 +511,34 @@ func (p *Post) manageStatusTransition() error {
 
 	emitEvent(e)
 
+	// If completed, hydrate CompletedOn. If not completed, nullify CompletedOn
+	// Don't use p.UpdateColumns, due to this being called by the AfterUpdate function
+	switch p.Status {
+	case PostStatusCompleted:
+		if !p.CompletedOn.Valid {
+			err := DB.RawQuery(
+				fmt.Sprintf(`UPDATE posts set completed_on = '%s' where ID = %v`,
+					time.Now().Format(domain.DateFormat), p.ID)).Exec()
```

Comment:
I believe Pop takes care of the format from a `Time` even in a RawQuery.
```suggestion
					time.Now(), p.ID)).Exec()
```

Reply:
That change breaks my tests.
```unable to set Post.CompletedOn for ID: 29, pq: invalid input syntax for type date: "2020-02-14 13:55:40.492484744 +0000 UTC m=+29.406122424"```

Reply:
Sorry, I Guess I remembered wrong. Maybe it was on a SELECT, though I would be surprised if that was different.